pkg/ciliumenvoyconfig/cell.go

```diff
@@ -169,8 +154,8 @@ type managerParams struct {
 	BackendSyncer  *envoyServiceBackendSyncer
 	ResourceParser *cecResourceParser
 
-	Services  allService
-	Endpoints allEndpoint
+	Services  resource.Resource[*slim_corev1.Service]
```

Comment:
i assume that you double-checked whether the providers of these k8s resources really watch and return ALL services/endpoints

Reply:
I've discussed this before with @sayboras , and my previous understanding was that it was introduced for headless services as the previous informer did not watch headless services at all.

To be more specific, this informer does not watch EndpointSlices:
- For Headless services created by Clustermesh - https://docs.cilium.io/en/latest/network/clustermesh/services/#endpointslicesync
- with `service.kubernetes.io/service-proxy-name` label if `k8s-service-proxy-name` / `k8s.serviceProxyName` not specified (default)

For Services, similar with `service.kubernetes.io/service-proxy-name` label. 
Do you think this is fine or what would be the reason that we have to watch ALL services/endpoints?

Reply:
Let's add @sayboras then - who was part of the latest refactoring (with headless service)

Reply:
Thanks for the ping both. 

Yes, the main reason of separate informers are for headless service.